### PR TITLE
Fix AddAssignment bugs

### LINKED ISSUE #71: assign does not throw proper error here
image
will do a quick fix to use parseAvailability instead of creating availability object.

should also create a parser to parse assignmentdetails

### LINKED ISSUE #72: assign is able to take multiple keywords
assign index d/details a/availability is the intended command entered.
But the current implementation allows assign 3 d/test1 d/test2 a/01/01/2024 a/02/02/2024. 
This results in the parser taking in the last entered value only.

### LINKED ISSUE #77: Find command does not redirect after lista
At the assignments page, after executing find command, the list does not redirect to person's list

### PR DESCRIPTION
**Pull request by Benson15912**
Fix #71 
Fix #72
Fix #77

Minor fixes to parser.

List of bugs fixed:
1.assign index d/details a/date
  a/date does not throw the invalid date format error. and causes illegalargumentexception

2.assign being able to take multiple keywords.

3.find command does not redirect to person's list while at assignment page

4.no parseDetails method in parseUtil

### CODE REVIEW

File: src/test/java/seedu/address/logic/parser/AddressBookParserTest.java [hunk 1]
```diff
@@ -114,7 +114,7 @@ public void parseCommand_viewAssignment() throws Exception {
     @Test
     public void parseCommand_addAssignment() throws Exception {
         String addAssignmentCommand = AddAssignmentCommand.COMMAND_WORD + " 1 " + PREFIX_DETAILS
-                + "VALID_DETAILS " + PREFIX_AVAIL + "01/01/2024";
+                + "VALIDDETAILS " + PREFIX_AVAIL + "01/01/2024";
```
Comment (SimKianSeng) [inline src/test/java/seedu/address/logic/parser/AddressBookParserTest.java:117]: Should the variable name be `VALID_DETAILS` or `VALIDDETAILS`? Perhaps a typo in variable name?

**Review by SimKianSeng — APPROVED** [review]
Overall LGTM, minor style issue? Can take a look before merging.

**Review by SimKianSeng — APPROVED** [review]
LGTM

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 2]
```diff
@@ -107,6 +108,19 @@ public static Set<Availability> parseAvailabilities(Collection<String> availabil
         return availabilitiesSet;
     }
 
+    /**
+     * Parses {@code String Details} into a {@code AssignmentDetails}.
+     */
+    public static AssignmentDetails parseAssignmentDetails(String details) throws ParseException {
+        requireNonNull(details);
+        String trimmedDetails = details.trim();
+        if (!AssignmentDetails.isValidAssignmentDetails(trimmedDetails)) {
```
Comment (xuwen-ho) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:117]: Good defensive programming

**Review by xuwen-ho — APPROVED** [review]
Works well and fixes bug

**Pull request merged**